libs/auth/src/dipped_smart_card_auth_with_memory_card.ts

```diff
@@ -161,24 +151,31 @@ export class DippedSmartCardAuthWithMemoryCard
     return ok();
   }
 
-  setElectionDefinition(electionDefinition: ElectionDefinition): void {
-    this.config.electionDefinition = electionDefinition;
-  }
-
-  clearElectionDefinition(): void {
-    delete this.config.electionDefinition;
+  private async checkCardReaderAndUpdateAuthStatus(
+    machineState: DippedSmartCardAuthMachineState
+  ) {
+    const cardSummary = await this.card.readSummary();
+    this.updateAuthStatus(machineState, {
+      type: 'check_card_reader',
+      cardSummary,
+    });
```

Comment:
This is quite an interesting piece of code after this change. It's an auth state manager that only changes state when you check it. Quite Schroedinger's Cat-like. I don't see any problem with it, but I don't know if I've seen something quite like this before, so it makes me slightly wary.

Reply:
The Schroedinger's Cat analogy really works 😆

Yeah, I felt this too. I couldn't conceive of a reasonable way to compute the auth status before it was requested without returning to some form of election def caching in the auth lib.

The main risk that I see with this new approach is forgetting to refresh auth status internally before performing another auth status changing operation, like PIN checking. But at least that risk is contained to the auth lib, unlike the risk of correctly caching the election def, which would affect all consumers.

So given the smaller risk surface area of this new approach, I do think I prefer it. Will think of ways to mitigate the risk of forgetting to refresh auth status internally before performing another auth status changing operation when I create the Java Card implementations!

Reply:
I'll also be curious to see whether the need to always provide election hash to auth methods becomes cumbersome, e.g. would this complicate creating an auth decorator? (I don't think so but just noting)

Reply:
The reasoning about reducing the risk surface area makes sense to me!